Vector2.h

```diff
@@ -37,6 +37,12 @@ class Vector2
 	{
 		return sqrtf(x * x + y * y);
 	}
+
+	// Returns the vector scaled to fit within a unit vector
+	Vector2 normalized()
+	{
+		return Vector2(this->x / this->magnitude(), this->y / this->magnitude());
```

Comment:
Slight inefficiency -- you're calculating `magnitude()` twice here.

Where you use this on line 97 of Application.cpp, you're similarly calling it twice -- so in that case you're doing four calls to `magnitude()` where only one is needed!